ophyd/areadetector/plugins.py

```diff
@@ -232,6 +244,29 @@ def _asyn_pipeline_configuration_names(self):
     time_stamp = C(EpicsSignalRO, 'TimeStamp_RBV')
     unique_id = C(EpicsSignalRO, 'UniqueId_RBV')
 
+    @plugin_type.sub_connect
+    def _plugin_type_connected(self, connected, **kw):
+        'Connection callback on the plugin type'
+        if not connected:
+            return
+
+        plugin_type = self.plugin_type.get()
+        self._misconfigured = not plugin_type.startswith(self._plugin_type)
+        if self._misconfigured:
+            logger.warning(
+                'Plugin prefix %r: trying to use %r class (plugin type=%r) '
+                ' but the plugin reports it is of type %r',
+                self.prefix, self.__class__.__name__, self._plugin_type,
+                plugin_type
+            )
```

Comment:
Any worry about the change in behavior here? The previous setup would raise an exception on init, while this logs a warning instead. I think this is better but are we concerned about changing it?

Reply:
Good question - my opinion is that the old behavior was annoying enough to warrant this change. 

This will `warn` and won't allow you to `stage` the detector, which is probably the most important portion. What does everyone else think?